### PR TITLE
Simplify the code to find best values.

### PR DESCRIPTION
**Pull request by c-bata**
## Motivation

Simplify the code to find best values.

## Description of the changes

```python
>>> import numpy as np
>>> values = [5, 3, 4, 2, 2, 3, 1]
>>> np.minimum.accumulate(values)
array([5, 3, 3, 2, 2, 2, 1])
>>> 
```

### CODE REVIEW

**Review by keisuke-umezawa — APPROVED** [review]
LGTM!

**Comment by keisuke-umezawa** [comment]
@not522 If you have time, could you look on it?

File: optuna/visualization/_optimization_history.py [hunk 1]
```diff
@@ -91,15 +93,9 @@ def _get_optimization_history_plot(
 
     if target is None:
         if study.direction == StudyDirection.MINIMIZE:
-            best_values = [float("inf")]
+            best_values = list(np.minimum.accumulate([t.value for t in trials]))
```
Comment (not522) [inline optuna/visualization/_optimization_history.py:96]:
```suggestion
            best_values = np.minimum.accumulate([t.value for t in trials])
```

File: optuna/visualization/_optimization_history.py [hunk 2]
```diff
@@ -91,15 +93,9 @@ def _get_optimization_history_plot(
...
         else:
-            best_values = [-float("inf")]
-        comp = min if study.direction == StudyDirection.MINIMIZE else max
-        for trial in trials:
-            trial_value = trial.value
-            assert trial_value is not None  # For mypy
-            best_values.append(comp(best_values[-1], trial_value))
-        best_values.pop(0)
+            best_values = list(np.maximum.accumulate([t.value for t in trials]))
```
Comment (not522) [inline optuna/visualization/_optimization_history.py:98]:
```suggestion
            best_values = np.maximum.accumulate([t.value for t in trials])
```

File: optuna/visualization/matplotlib/_optimization_history.py [hunk 3]
```diff
@@ -97,13 +99,10 @@ def _get_optimization_history_plot(
...
 
-    best_values = [float("inf")] if study.direction == StudyDirection.MINIMIZE else [-float("inf")]
-    comp = min if study.direction == StudyDirection.MINIMIZE else max
-    for trial in trials:
-        trial_value = trial.value
-        assert trial_value is not None  # For mypy
-        best_values.append(comp(best_values[-1], trial_value))
-    best_values.pop(0)
+    if study.direction == StudyDirection.MINIMIZE:
+        best_values = list(np.minimum.accumulate([t.value for t in trials]))
```
Comment (not522) [inline optuna/visualization/matplotlib/_optimization_history.py:103]:
```suggestion
        best_values = np.minimum.accumulate([t.value for t in trials])
```

File: optuna/visualization/matplotlib/_optimization_history.py [hunk 4]
```diff
@@ -97,13 +99,10 @@ def _get_optimization_history_plot(
...
+    else:
+        best_values = list(np.maximum.accumulate([t.value for t in trials]))
```
Comment (not522) [inline optuna/visualization/matplotlib/_optimization_history.py:105]:
```suggestion
        best_values = np.maximum.accumulate([t.value for t in trials])
```

**Review by not522 — CHANGES_REQUESTED** [review]
Thank you! It seems `np.minimum.accumulate([t.value for t in trials])` is enough in this case.

**Comment by c-bata** [comment]
Thank you! I applied your suggestions.

**Comment by not522** [comment]
Basically LGTM. Please update unit test to pass CI.

**Comment by c-bata** [comment]
Thank you. I fixed test cases.

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**